src/AltinnCore/Common/Helpers/InstantiationHelper.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using Altinn.Platform.Storage.Models;
+using AltinnCore.ServiceLibrary.Enums;
+using AltinnCore.ServiceLibrary.Models;
+
+namespace Common.Helpers
+{
+    /// <summary>
+    /// Various helper methods for instantiation
+    /// </summary>
+    public static class InstantiationHelper
+    {
+        /// <summary>
+        /// Filters a list of parties based on an applications allowed party types.
+        /// </summary>
+        /// <param name="parties">The list of parties to be filtered</param>
+        /// <param name="partyTypesAllowed">The allowed party types</param>
+        /// <returns>A list with the filtered parties</returns>
+        public static List<Party> FilterPartiesByAllowedPartyTypes(List<Party> parties, PartyTypesAllowed partyTypesAllowed)
```

Comment:
the class name PartyTypesAllowed sets a restriction on what the class can be used for. it's basically a list of party types that we could use for other things as well.. maybe consider renaming?

Reply:
Do you have any usecases for this? Renaming this would require us to publish a new nuget package with the updated name. If there are no concrete examples I feel like this is over engineering.

Reply:
Just an FYI, the PartyTypesAllowed object is part of the Application metadata object.